deploy/gk-deploy

```diff
@@ -572,107 +590,112 @@ if [[ ${LOAD} -eq 0 ]]; then
+  if [[ $("${heketi_cli}" volume list 2>&1) != *heketidbstorage* ]]; then
+    eval_output "${heketi_cli} setup-openshift-heketi-storage 2>&1"
+    if [[ ${?} != 0 ]]; then
+      output "Failed on setup openshift heketi storage"
+      exit 1
+    fi
+  else
+    output "Volume heketidbstorage not found."
     exit 1
   fi
-else
-  output "Volume heketidbstorage not found."
-  exit 1
-fi
 
-eval_output "${CLI} exec -it ${heketi_pod} -- cat heketi-storage.json | ${CLI} create -f - 2>&1"
-if [[ ${?} != 0 ]]; then
-  output "Failed on creating heketi storage resources."
-  exit 1
-fi
+  eval_output "${CLI} exec -it ${heketi_pod} -- cat heketi-storage.json | ${CLI} create -f - 2>&1"
+  if [[ ${?} != 0 ]]; then
+    output "Failed on creating heketi storage resources."
+    exit 1
+  fi
 
-check_pods "job-name=heketi-storage-copy-job" "Completed"
-if [[ ${?} != 0 ]]; then
-  output "Error waiting for job 'heketi-storage-copy-job' to complete."
-  exit 1
-fi
+  check_pods "job-name=heketi-storage-copy-job" "Completed"
+  if [[ ${?} != 0 ]]; then
+    output "Error waiting for job 'heketi-storage-copy-job' to complete."
+    exit 1
+  fi
 
-eval_output "${CLI} label --overwrite svc heketi-storage-endpoints glusterfs=heketi-storage-endpoints heketi=storage-endpoints"
-eval_output "${CLI} delete all,service,jobs,deployment,secret --selector=\"deploy-heketi\" 2>&1"
-if [[ "${CLI}" == *oc\ * ]]; then
-  eval_output "${CLI} delete dc,route,template --selector=\"deploy-heketi\" 2>&1"
+  eval_output "${CLI} label --overwrite svc heketi-storage-endpoints glusterfs=heketi-storage-endpoints heketi=storage-endpoints"
+  eval_output "${CLI} delete all,service,jobs,deployment,secret --selector=\"deploy-heketi\" 2>&1"
+  if [[ "${CLI}" == *oc\ * ]]; then
+    eval_output "${CLI} delete dc,route,template --selector=\"deploy-heketi\" 2>&1"
+  fi
 fi
 
-if [[ "${CLI}" == *oc\ * ]]; then
-  eval_output "${CLI} process ${OC_PROCESS_VAL_SWITCH} HEKETI_EXECUTOR=${EXECUTOR} ${OC_PROCESS_VAL_SWITCH} HEKETI_ADMIN_KEY=${ADMIN_KEY} ${OC_PROCESS_VAL_SWITCH} HEKETI_USER_KEY=${USER_KEY} heketi | ${CLI} create -f - 2>&1"
-else
-  eval_output "sed -e 's/\\\${HEKETI_EXECUTOR}/${EXECUTOR}/' -e 's/\\\${HEKETI_ADMIN_KEY}/${ADMIN_KEY}/' -e 's/\\\${HEKETI_USER_KEY}/${USER_KEY}/' ${TEMPLATES}/heketi-deployment.yaml | ${CLI} create -f - 2>&1"
-fi
+if [[ ${EXISTS_HEKETI} -eq 0 ]]; then
+  if [[ "${CLI}" == *oc\ * ]]; then
+    eval_output "${CLI} process ${OC_PROCESS_VAL_SWITCH} HEKETI_EXECUTOR=${EXECUTOR} ${OC_PROCESS_VAL_SWITCH} HEKETI_ADMIN_KEY=${ADMIN_KEY} ${OC_PROCESS_VAL_SWITCH} HEKETI_USER_KEY=${USER_KEY} heketi | ${CLI} create -f - 2>&1"
+  else
+    eval_output "sed -e 's/\\\${HEKETI_EXECUTOR}/${EXECUTOR}/' -e 's/\\\${HEKETI_ADMIN_KEY}/${ADMIN_KEY}/' -e 's/\\\${HEKETI_USER_KEY}/${USER_KEY}/' ${TEMPLATES}/heketi-deployment.yaml | ${CLI} create -f - 2>&1"
+  fi
 
-output -n "Waiting for heketi pod to start ... "
-check_pods "glusterfs=heketi-pod"
```

Comment:
This change check_pods "glusterfs=heketi-pod" --> check_pods "heketi=pod" looks like it should belong to the label cleanup patch...